File: python/a03.py
N, K = map(int, input().split())
P = list(map(int, input().split()))
Q = list(map(int, input().split()))

isFound = False
for i in range(N):
    for j in range(N):
        if P[i] + Q[j] == K:
            isFound = True
            break
if isFound:
    print("Yes")
else:
    print("No")

File: scala/a03.scala
import  scala.io.StdIn._

object Main {
    def main(args: Array[String]): Unit= {
        val Array(n,k) = readLine().split(" ").map(_.toInt)
        val listP = readLine().split(" ").map(_.toInt)
        val listQ = readLine().split(" ").map(_.toInt)
        val isFound = (for { p <- listP; q <- listQ } yield p + q ).filter(_ == k).length
        val ans = if (isFound > 0) "Yes" else "No"
        println(ans)
    }
}